native-sql-engine/core/src/main/scala/org/apache/spark/sql/execution/ColumnarShuffleExchangeExec.scala

```diff
@@ -86,7 +89,11 @@ case class ColumnarShuffleExchangeExec(
     // check input datatype
     for (attr <- child.output) {
       try {
-        ConverterUtils.createArrowField(attr)
+        if (!columnarConf.enableComplexType) {
+          ConverterUtils.checkIfTypeSupported(attr.dataType)
```

Comment:
@zhixingheyi-tian can you please also check if this make sense to you? This helps to easily disable(fallback) on complex type

Reply:
Make sense.